### ISSUE TITLE
support(http): add reproduction for issue #39285

### ISSUE THREAD

**Issue by bampakoa** [angularathens/2nd-workshop-angular-contrib#23]
Create an Angular application using StackBlitz that reproduces the issue in https://github.com/angular/angular/issues/39285.

**Comment by gkapris** [comment]
I would like to check this issue

**Comment by gkapris** [comment]
https://stackblitz.com/edit/angular-spread-non-iterable

Implementation of the issue in two ways.
1. the request in an service file
2. the request in an interceptor

In both implementations the code works fine.

I look forward to your comments

**Comment by gkalpak** [comment]
Thx for looking into this @gkapris. I also looked at the [source code][1] and it seems that the header `value` is always [coerced to an array][2] if it is a string. Based on that, I think we can go ahead and close angular/angular#39285 as non-reproducible.

Could you please post a comment with your reproduction on angular/angular#39285? 

[1]: https://github.com/angular/angular/blob/eac85b4e235fc99306945012a62ba4333e524718/packages/common/http/src/headers.ts
[2]: https://github.com/angular/angular/blob/eac85b4e235fc99306945012a62ba4333e524718/packages/common/http/src/headers.ts#L219-L221

**Comment by rohandani** [comment]
I have observed that the below issue 
"TypeError: Invalid attempt to spread non-iterable instance in Angular 10 with string http header"
happens from Angular 9+ versions if at all the value of the request header is something else that the string.
e.g. I was setting my request header as headers.set("user_id", 3) and started getting the above issue when I migrated my codebase to Angular10 but converting it to string solved the issue i.e.  headers.set("user_id", "3")
